### DISCUSSION TITLE
Adding a nova CRUD template

### DISCUSSION THREAD

**Discussion by sanzor** [novaframework/nova#217] [Ideas]
I think it would be a lot more intuitive to understand and use nova if it  could come with a template with the basic CRUD operations over a given domain model , so that the user could see how can he define the different exposed methods using either querystring or json.
Ideally the template could also contain the code to decode  (using a serializer) and fetch some fields of the json 
and also the code in order to encode back and send the json response.

The template could include a controller with the methods 
 - GET (querystring , responds in json) 
 - POST (receives json , responds in json)
 - PUT (receives json , responds in json)
 - DELETE (receives query string , responds with status or whatever)

```
-module(pet_controller).
-export([get/1, create/1, delete/1,update/1]).

get(Req=#{bindings := #{<<"id">> := Id}})->
   .....
   Pet= fetch_pet_from_somewhere(Id),
   Encode= thoas:encode(Pet),
   {json,200,#{},Encode}.

get(Req=#{parsed_qs:= #{<<"id">> := Id}})->
   .....
   Pet= fetch_pet_from_somewhere(Id),
   Encode= thoas:encode(Pet),
   {json,200,#{},Encode}.

create(Req=#{json := { <<"id">> := Id , <<"hair_color">> := HairColor } })->
    insert_to_db(Req),
     {json,200,#{},<<"created">>}.

update(Req=#{json := { <<"id">> := Id , <<"hair_color">> := HairColor } })->
    update_to_db(Req),
     {json,200,#{},<<"updated">>}.

delete(Req=#{parsed_qs:= { <<"id">> := Id } })->
    delete(Req),
     {json,200,#{},<<"deleted">>}.  
 
```
The datalayer could just be a mock or another module with undefined methods or just some constant in the method , it doesn't rlly matter but the important part is for the new user to see how he handles json/binding/querystring,  serialization,deserialization ,  json deconstruction (fetching particular fields)

And ideally this template could just be started and checked out with swagger.

P.S The idea came from C# where there are a couple of web api templates pic related
![image](https://github.com/novaframework/nova/assets/26228414/c5351caf-1ff7-4f62-ae3d-75e7fa492068)

**Comment by burbas** [comment]
Think this is worth re-visiting. A CRUD-template would make sense to have. Will look into it later today,

**Reply by sanzor** [reply]
From that time , i already published on article on how to create a Nova simple controller and it is also listed on nova docs :
https://bercovici-adrian-simon.medium.com/building-an-erlang-web-api-using-nova-framework-and-redis-141edf170ef7

**Comment by burbas** [comment]
Right. We have actually linked to that article from novaframework.org :)

**Reply by sanzor** [reply]
Feel free to add it, or chunk it and integrate it in some module/template if its of use.
